app/models/event/submission.rb

```diff
@@ -23,6 +23,11 @@ def event_json
             'id' => @assignment.submission_id,
             'created_at' => @assignment.updated_at,
             'content' => @assignment.solution,
-            'guide' => {'chapter' => @assignment.guide.chapter.as_json(only: [:id], methods: [:name])})
+            'guide' => { 'parent' => {
+                          'type' => @assignment.exercise.navigable_parent.class.to_s,
+                          'name' => @assignment.exercise.navigable_parent.try(:name),
+                          'position' => @assignment.exercise.navigable_parent.try(:number),
+                          'chapter' => @assignment.guide.chapter.as_json(only: [:id], methods: [:name])
+                        }})
```

Comment:
Here you are repeating `exercise.naviagable_parent.try` everywhere. 

You could build the json within the try block instead. E.g: 

```
'guide' => { 'parent' =>    @assignment.exercise.navigable_parent.try do |parent| 
                       {
                           'type' => parent.class.to_s,
                           'name' =>parent.name,
                           'position' => parent.number,
                           ...
                         }
end})
```